### PR TITLE
Add the new media flow to the media text block

### PR DESCRIPTION
**Pull request by draganescu**
## Description
Requires #18139 There is a new replace flow offered by a component which is currently implemented in the Image Block. This PR implements the new flow in the Media + Text block.

## How has this been tested?
Tested locally.

## Screenshots 

Screenshot 2019-12-17 at 17 06 37

### CODE REVIEW

File: packages/block-library/src/media-text/media-container.js [hunk 1]
```diff
@@ -45,24 +45,16 @@ class MediaContainer extends Component {
...
-								icon="edit"
-								onClick={ open }
-							/>
-						) }
-					/>
-				</ToolbarGroup>
+				<MediaReplaceFlow
+					mediaURL={ mediaUrl }
+					allowedTypes={ ALLOWED_MEDIA_TYPES }
+					accept="image/*"
```
Comment (Soean) [inline packages/block-library/src/media-text/media-container.js]: We also need to be able to upload video files.

File: packages/block-editor/src/components/index.js [hunk 2]
```diff
@@ -45,6 +45,7 @@ export { default as ToolSelector } from './tool-selector';
 export { default as URLInput } from './url-input';
 export { default as URLInputButton } from './url-input/button';
 export { default as URLPopover } from './url-popover';
+export { __experimentalImageURLInputUI } from './url-popover/image-url-input-ui';
```
Comment (Soean) [inline packages/block-editor/src/components/index.js]: Do we need to include the `ImageURLInputUI` change here or can we move it into a separate PR?

**Review by Soean — CHANGES_REQUESTED** [review]

File: packages/block-library/src/media-text/media-container.js (hunk 1 above)
Comment (draganescu) [reply packages/block-library/src/media-text/media-container.js]: howdy @Soean fixed this!

File: packages/block-editor/src/components/index.js (hunk 2 above)
Comment (draganescu) [reply packages/block-editor/src/components/index.js]: I think this was because the branch needed a rebase.

File: packages/block-library/src/media-text/edit.js [hunk 3]
```diff
@@ -117,6 +119,30 @@ class MediaTextEdit extends Component {
...
+			// Check if the image is linked to the attachment page.
+			if ( linkDestination === LINK_DESTINATION_ATTACHMENT ) {
+				// Update the media link.
+				newHref = undefined;
+			}
+			this.props.setAttributes( {
+				mediaUrl: newURL,
+				href: newHref,
+				mediaId: undefined,
+				sizeSlug: DEFAULT_SIZE_SLUG,
```
Comment (Soean) [inline packages/block-library/src/media-text/edit.js]: There is no `sizeSlug` attribute in the media-text block.

File: packages/block-library/src/media-text/edit.js [hunk 4]
```diff
@@ -43,8 +43,9 @@ const TEMPLATE = [
 const WIDTH_CONSTRAINT_PERCENTAGE = 15;
 const applyWidthConstraints = ( width ) => Math.max( WIDTH_CONSTRAINT_PERCENTAGE, Math.min( width, 100 - WIDTH_CONSTRAINT_PERCENTAGE ) );
 
-export const LINK_DESTINATION_MEDIA = 'media';
-export const LINK_DESTINATION_ATTACHMENT = 'attachment';
+const LINK_DESTINATION_MEDIA = 'media';
+const LINK_DESTINATION_ATTACHMENT = 'attachment';
+const DEFAULT_SIZE_SLUG = 'large';
```
Comment (Soean) [inline packages/block-library/src/media-text/edit.js]: We can remove the default, because there is no `sizeSlug` attribute.

**Review by Soean — CHANGES_REQUESTED** [review]

**Review by Soean — APPROVED** [review]
Thanks, works great!
